### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by Apanatshka**
Writing tests as described in #3. 
Going straight for the bonus points: so far I've written quickcheck tests for a number of unit test situations. 
# TODO list
## Unit tests
- [X] Test the benchmark parser, including the fact that throughput is optional.
- [X] Test that find overlap finds the correct pairs.
- [X] Test that find overlap reports correct missing benchmarks (for both old and new).
- [x] Test that the name determination for the two columns works correctly when given two file paths. (This may require refactoring the `names` method off of the `Args` struct.)
- [X] Test `commafy`.
- [x] Test that the name prefix feature works. (Refactor code to get a unit test.)
## Integration tests
- [x] Test some basic example inputs and check that it produces the expected output.
- [x] Test that passing input on stdin works.

### CODE REVIEW

**Comment by Apanatshka** [comment]
@BurntSushi: I refactored some code to make it more testable, but didn't do the most minimal change because I saw room for improvement. The changes are in a2caf903313055c7850fb0268b585ae46e734ead.

**Comment by Apanatshka** [comment]
The last two test can be implemented once uutils/coreutils#959 is resolved, which should provide a nice crate for integration testing of command line tools.

**Comment by Apanatshka** [comment]
Looks like the appveyor build is failing because of https://github.com/rust-lang/rust/issues/35991.

**Comment by Apanatshka** [comment]
@BurntSushi: Ready for review/merge

**Comment by Apanatshka** [comment]
@BurntSushi Still waiting on a review...

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,11 @@ license = "Unlicense/MIT"
 [dependencies]
 docopt = "0.6.81"
 lazy_static = "0.2.1"
-prettytable-rs = "0.6.2"
+quickcheck = "0.3.1"
```
Comment (BurntSushi) [inline Cargo.toml]: Both `quickcheck` and `second_law` should be `dev-dependencies`, which means they'll only be required during tests.

**Comment by BurntSushi** [comment]
@Apanatshka Overall this looks really good. Could you say whether there are any behavior changes here?

Sorry about neglecting your PRs. I've been really into another project, and any leftover time I had after that has gone into wedding planning!

**Comment by Apanatshka** [comment]
## In reverse order:

> I've been really into another project,

That doesn't surprise me at all, you seem to be quite the collector :P 

> and any leftover time I had after that has gone into wedding planning!

If that's not a legitimate "real life" excuse, I don't know what is :) Good luck with that, and remember to enjoy it rather than stressing out ^^

> Sorry about neglecting your PRs

I don't mind nagging until you respond if you don't ;)

> Could you say whether there are any behavior changes here?

Only one outside observable change that I'm aware of: Newlines in the output are now consistently unix `\n`. I found during testing that `prettytable-rs` actually did OS-specific line breaks, but in our own output code that wasn't done. The OS-specific stuff is now optional through a feature flag in that crate, so I turned it off.

**Comment by BurntSushi** [comment]
All right great! Since I merged your other color PR, this one now has a conflict. After a rebase I'll merge. :-)

**Comment by Apanatshka** [comment]
Hah, the test for the usage string just failed, because the `--color` option is now added. 

Question: Is this even a good test? I basically duplicated the string in the code to a file and test that that's the output on `-h`/`--help`...

**Comment by BurntSushi** [comment]
@Apanatshka Oh, wow, I see. I missed that. I'd throw away that test. It's better to just write new tests for each new flag/option added. I think having to update a separate file every time the usage string changes is probably too annoying.

**Comment by Apanatshka** [comment]
Ok then. The usage string test is removed now. 
In a related category, there is still a test in there for the version flag. But it grabs the version from a cargo environment variable, so it shouldn't be fragile. 

> It's better to just write new tests for each new flag/option added. 

Yeah, ideally I'd like to add integration tests for the coloured output.. But the planned "quickly rebase so it can be merged today" is already taking more time that I expected. I should really get back to work now ^^' In 6 hours I'll have time to capture the output with terminal colour codes in a fixture for a test.

**Comment by BurntSushi** [comment]
@Apanatshka This is great, thank you!

(Tests for coloring are strictly nice to have. Don't sweat it. :-))

**Pull request merged**